autoscale/src/test/java/com/sequenceiq/periscope/endpointtests/DistroXAutoScaleScalingActivityV1EndpointTest.java

```diff
@@ -0,0 +1,283 @@
+
+        Cluster testCluster = new Cluster();
+        testCluster.setStackCrn(TEST_CLUSTER_CRN);
+        testCluster.setStackName(TEST_CLUSTER_NAME);
+        testCluster.setCloudPlatform("AWS");
+
+        ClusterPertain clusterPertain = new ClusterPertain();
+        clusterPertain.setTenant(TEST_ACCOUNT_ID);
+        clusterPertain.setWorkspaceId(TEST_WORKSPACE_ID);
+        clusterPertain.setUserId(TEST_USER_ID.toString());
+        clusterPertain.setUserCrn(TEST_USER_CRN);
+
+        UserManagementProto.User user = UserManagementProto.User.newBuilder()
+                .setCrn(TEST_USER_CRN).setEmail("dummyuser@cloudera.com").setUserId(TEST_USER_ID.toString()).build();
+        testCluster.setClusterPertain(
+                clusterPertainRepsitory.findFirstByUserCrn(clusterPertain.getUserCrn())
+                        .orElseGet(() -> clusterPertainRepsitory.save(clusterPertain)));
+        clusterRepository.save(testCluster);
+        when(grpcUmsClient.getUserDetails(anyString(), any())).thenReturn(user);
+        doNothing().when(resourceAuthorizationService).authorize(eq("crn:cdp:iam:us-west-1:accid:cluster:mockuser@cloudera.com"), any(), any());
+        when(clusterProxyConfigurationService.getClusterProxyUrl()).thenReturn(Optional.of("http://clusterproxy"));
+    }
+
+    @After
+    public void tearDown() {
+        clusterRepository.deleteAll();
+        clusterPertainRepsitory.deleteAll();
+    }
+
+    @Test
+    public void testGetAllScalingActivityUsingClusterName() {
+        Cluster testCluster = new Cluster();
+        testCluster.setStackCrn(TEST_CLUSTER_CRN);
+        testCluster.setStackName(TEST_CLUSTER_NAME);
+        testCluster.setCloudPlatform("AWS");
+
+        ScalingActivity scalingActivity = createScalingActivity(testCluster, ActivityStatus.METRICS_COLLECTION_SUCCESS, Instant.now().toEpochMilli());
+        doReturn(List.of(scalingActivity)).when(scalingActivityService).findAllForCluster(any(), any());
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getAllScalingActivityByClusterName(testCluster.getStackName(), 0, 1)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of(scalingActivity.getActivityCrn()));
+    }
+
+    @Test
+    public void testGetAllScalingActivityUsingClusterCrn() {
+        Cluster testCluster = new Cluster();
+        testCluster.setStackCrn(TEST_CLUSTER_CRN);
+        testCluster.setStackName(TEST_CLUSTER_NAME);
+        testCluster.setCloudPlatform("AWS");
+
+        ScalingActivity scalingActivity = createScalingActivity(testCluster, ActivityStatus.METRICS_COLLECTION_SUCCESS, Instant.now().toEpochMilli());
+        doReturn(List.of(scalingActivity)).when(scalingActivityService).findAllForCluster(any(), any());
```

Comment:
Same here and for all stubs below - Please check if stubs actually need to be added

Reply:
Done